commons/src/main/java/io/aiven/kafka/tieredstorage/commons/SegmentCompressionChecker.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.FileRecords;
+
+/**
+ * Checks if segment requires compression or not.
+ * To be used when segment files are received on archival.
+ */
+public class SegmentCompressionChecker {
+
+    /**
+     * @param file Kafka log segment file
+     * @return true if requires compression, else false
+     */
+    public static boolean check(final File file) throws IOException {
+        try (final FileRecords records = FileRecords.open(file)) {
+            return records.firstBatch().compressionType() == CompressionType.NONE;
```

Comment:
I think we also need to call `ensureValid()` on the batch.

It throws `CorruptRecordException` in case the batch is invalid. We need to think how to propagate it further (it's unchecked).

Reply:
Not sure about calling `ensureValid` at this point. It will validate CRC introducing additional overhead, while Compression type is on the record batch headers already. 
If we are looking to ensure a log segment is valid, we could do it outside this compression check and pass FileRecords instead of File as input, wdyt?

Reply:
> If we are looking to ensure a log segment is valid

No, that's not the point. The idea is that since we're reading from disk, any corruption is possible and if the batch is invalid, `compressionType` may contain nonsense (may be just incorrect or throw `IllegalArgumentException` because the value doesn't match anything known). `ensureValid` is the way to ensure the batch integrity with good certainty.
I wouldn't bother of the overhead also, CRC32 calculation is fast and this segment will reach the page cache anyway.

Reply:
Sure, fair enough. Giving a try now, lmk how it looks